R/as-data-frame.R

```diff
@@ -73,3 +73,49 @@ as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE
     names(out) <- names(x)
     return(structure(out, class="data.frame", row.names=c(NA, -nrow(x))))
 }
+
+
+#' as.data.frame method for VariableCatalog
+#'
+#' This method allows you to import the VariableCatalog into your R sessions in 
+#' order to facilliate interactive use. Modifying the dataframe produced by this
+#' method will not update the Crunch web app, for information about modifying 
+#' variables in the app see: \code{vingette("variables",
+#' package = "crunch")}  
+#' 
+#' @param x
+#' A VariableCatalog produced by \code{variables(ds)}.
+#' @param ... Additional arguments passed to \code{data.frame}
+#' @param row.names part of as.data.frame signature. Ignored.
+#' @param optional part of as.data.frame signature. Ignored.
+#' @param fields 
+#' A character vector of the fields from the variable catalog which you would like 
+#' included in the datafame. To include all fields or see which fields are available
+#' set fields to "all". 
+#' @return A data frame
+#' @examples 
+#' \dontrun{
+#' ds <- loadDataset("iris")
+#' vars <- variables(ds)
+#' var_df <- as.data.frame(vars, fields = "all")
+#' }
+#' 
+#' @rdname VariableCatalog-to-Data-Frame
+#' @export
+as.data.frame.VariableCatalog <- function(x, 
+    row.names = NULL, 
+    optional = FALSE, 
+    ...,
+    fields = c("alias", "name", "type")) {
+    out <- catalogToDataFrame(x, rownames = NA)
```

Comment:
I made some changes to the error message can't figure out how to replicate the current behaviour at the lower level. Currently the error message does two things:
1) Checks whether the field names are present in the dataframe; and
2) Suggests fields in the dataframe which aren't present in the requested fields list. 

So if someone asks for "banana" it will say something like "there is no field called 'banana', did you mean 'alias' ..." Now I can replicate the first bit at the `catalogToDataframe()` level by passing the filed name as the `keys` argument, and then checking the resulting data against the requested fields. I don't think I'll be able to provide guidance on available fields without effectively doing what I'm doing now. 

More generally, are we sure we want the same error behaviour for all current and future catalog types? I could envision a scenario where you might want to ask for fields which aren't present in the catalog and not expect an error, so from that perspective it might make more sense to have the `catalogToDataframe()` function just handle pulling the data into a data frame, and then have the `as.data.frame()` methods handle the error checking.